eva/readers/abstract_reader.py

```diff
@@ -52,7 +52,10 @@ def read(self) -> Iterator[Batch]:
         row_size = None
         for data in self._read():
             if row_size is None:
-                row_size = data['data'].nbytes
+                if 'data' in data:
+                    row_size = data['data'].nbytes
```

Comment:
Can we use `from sys import getsizeof` to estimate? So it works for general tables.

Reply:
Not sure if it provides the size of nested objects. I remember testing something similar but it didn't give the correct size. We need to check if there is a better way to do it.

Reply:
ya get sizeof does not seem to return correct values for nested dictionaries.

Reply:
I see. Will we have the nested dictionaries?